src/aiodynamo/http/httpx.py

```diff
@@ -1,19 +1,38 @@
 import json
 from dataclasses import dataclass
+from functools import wraps
 from typing import Any, Dict, Optional
 
 from aiodynamo.types import Timeout
-from httpx import AsyncClient
+from httpx import AsyncClient, HTTPError
 from yarl import URL
 
 from ..errors import exception_from_response
 from .base import HTTP, Headers, RequestFailed
 
 
+def wrap_errors(coro):
+    @wraps(coro)
+    async def wrapper(*args, **kwargs):
+        try:
+            return await coro(*args, **kwargs)
+        except HTTPError as exc:
+            raise RequestFailed(
+                exc.request and exc.request.url,
+                exc.response and exc.response.status_code,
+                exc.response and await exc.response.aread(),
+                exc.request and exc.request.headers,
+                exc.request and await exc.request.aread(),
```

Comment:
Request is ours, so I guess it should be OK?
Do we want to the body for anything though, are we logging it? Is it safe to log, i.e. there are no keys there?